### PR TITLE
RSDEV-271: Create adapter for Digital Commons Data integration

### PR DESCRIPTION
**Pull request by rs-nicof**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    </dependency>
+    <dependency>
+      <groupId>com.github.rspace-os</groupId>
+      <artifactId>rspace-repository-spi</artifactId>
+      <version>1.0.1</version>
+    </dependency>
+    <dependency>
+      <groupId>com.github.rspace-os</groupId>
+      <artifactId>digital-commons-data-java-client</artifactId>
+      <version>RSDEV-271-client-SNAPSHOT</version>
```
Comment (rs-nicof) [inline pom.xml]: I'll change this dependency version to be `0.0.1` once the [client PR](https://github.com/rspace-os/digital-commons-data-java-client/pull/1/) is merged

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,8 @@
+#!groovy
+@Library('rspace-shared') _ 
+// builds DigitalCommonsData-adapter project
+genericJavaLibBuild("emails":"dev@researchspace.com",
+ "branch":"${BRANCH_NAME}",
+  "jdk":"OPEN-JDK-11",
+  "maven":"maven3.8.1",
+  "jvmArgs":"-Djava-vendor=openjdk -Djava-version=11")
```
Comment (mKowalski256) [inline Jenkinsfile]: specifying java 11 shouldn't be necessary - if params are not provided, then[ Java 17 would be used by default](https://github.com/ResearchSpace-ELN/jenkins-pipeline-shared/blob/master/vars/genericJavaLibBuild.groovy), which is what we want here. Compare with [rspace-figshare-adapter](https://github.com/rspace-os/rspace-figshare-adapter/blob/main/Jenkinsfile).

File: src/main/java/com/researchspace/dcd/rspaceadapter/DigitalCommonsDataRepository.java [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+          "RestClientException occurred while submitting to DigitalCommonsData", null);
+    } catch (MalformedURLException e) {
+      log.error("MalformedURLException occurred while submitting to DigitalCommonsData", e);
+      return new RepositoryOperationResult(false,
+          "MalformedURLException occurred while submitting to DigitalCommonsData", null);
+    } catch (IOException e) {
+      log.error("IOException occurred while submitting to DigitalCommonsData", e);
+      return new RepositoryOperationResult(false,
+          "IOException occurred while submitting to DigitalCommonsData", null);
+    }
```
Comment (mKowalski256) [inline src/main/java/com/researchspace/dcd/rspaceadapter/DigitalCommonsDataRepository.java]: a lot of repeated code - would be more readable with a `catch-multiple` syntax

File: src/main/java/com/researchspace/dcd/utils/DigitalCommonsDataAdapterUtils.java [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class DigitalCommonsDataAdapterUtils {
+
+  public static List<License> getDigitalCommonsDataLicenses() throws MalformedURLException {
+    License license = new License();
+    license.setLicenseDefinition(new LicenseDef(new URL("https://creativecommons.org/publicdomain/zero/1.0/"), "CC-0"));
+    return Collections.singletonList(license);
+  }
```
Comment (mKowalski256) [inline src/main/java/com/researchspace/dcd/utils/DigitalCommonsDataAdapterUtils.java]: Seems a bit unnecessary to have an extra class with one method? I'd just put the code in `DigitalCommonsDataRepository`.

File: src/main/java/com/researchspace/dcd/rspaceadapter/DigitalCommonsDataRepository.java [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+  }
+
+  private DigitalCommonDataSubmission createDatasetRequest(SubmissionMetadata submissionMetadata) {
+    return DigitalCommonDataSubmission.builder()
+        .title(submissionMetadata.getTitle())
+        .description(submissionMetadata.getDescription())
+        .uploadType("export from RSpace")
```
Comment (mKowalski256) [inline src/main/java/com/researchspace/dcd/rspaceadapter/DigitalCommonsDataRepository.java]: I'm not sure what `uploadType` means in DCD, but "export from RSpace" sounds more like a description/message?

**Review by mKowalski256 — APPROVED** [review]
I've added a few comments, but in general the code looks good to me.

File: src/main/java/com/researchspace/dcd/rspaceadapter/DigitalCommonsDataRepository.java (hunk 5 above)
Comment (rs-nicof) [reply src/main/java/com/researchspace/dcd/rspaceadapter/DigitalCommonsDataRepository.java:82]: sure i'll leave as empty string for now

File: src/main/java/com/researchspace/dcd/utils/DigitalCommonsDataAdapterUtils.java (hunk 4 above)
Comment (rs-nicof) [reply src/main/java/com/researchspace/dcd/utils/DigitalCommonsDataAdapterUtils.java]: removed

File: Jenkinsfile (hunk 2 above)
Comment (rs-nicof) [reply Jenkinsfile]: changed

File: src/main/java/com/researchspace/dcd/rspaceadapter/DigitalCommonsDataRepository.java (hunk 3 above)
Comment (rs-nicof) [reply src/main/java/com/researchspace/dcd/rspaceadapter/DigitalCommonsDataRepository.java:75]: grouped now

File: src/main/java/com/researchspace/dcd/rspaceadapter/DigitalCommonsDataRepository.java (hunk 5 above)
Comment (rs-nicof) [reply src/main/java/com/researchspace/dcd/rspaceadapter/DigitalCommonsDataRepository.java]: as per our conversatino I'm changing it to "Exported from RSpace ELN"

**Pull request merged**